cpp/src/arrow/json/parser.cc

```diff
@@ -46,6 +46,10 @@ namespace arrow {
 using internal::BitsetStack;
 using internal::checked_cast;
 
+static const char* kEnvJsonMaxNumRows = "ARROW_JSON_MAX_NUM_ROWS";
```

Comment:
Instead of exposing this as an environment variable, please add it to ReadOptions or ParseOptions.